libraries/APM_Control/AP_RollController.cpp

```diff
@@ -202,13 +202,26 @@ float AP_RollController::get_servo_out(int32_t angle_err, float scaler, bool dis
     angle_err_deg = angle_err * 0.01;
     float desired_rate = angle_err_deg/ gains.tau;
 
+    /*
+      prevent indecision in the roll controller when target roll is
+      close to 180 degrees from the current roll
+     */
+    const float indecision_threshold_deg = 160;
+    if (fabsf(angle_err_deg) > indecision_threshold_deg) {
+        if (desired_rate * last_desired_rate < 0) {
+            desired_rate = -desired_rate;
+        }
+    }
+
     // Limit the demanded roll rate
     if (gains.rmax_pos && desired_rate < -gains.rmax_pos) {
         desired_rate = - gains.rmax_pos;
     } else if (gains.rmax_pos && desired_rate > gains.rmax_pos) {
         desired_rate = gains.rmax_pos;
     }
 
+    last_desired_rate = desired_rate;
```

Comment:
This needs to be kept upto date in the `get_rate_out` method so it is still correct when we stop VTOL or acro. I think you may actually be able to remove it all together and use `pinfo.target`which is set here:

https://github.com/ArduPilot/ardupilot/blob/f328a659ef6a0acea80c7fb16c2c621fb1f3be15/libraries/APM_Control/AP_FW_Controller.cpp#L84

Reply:
fixed